lib/markdownlint.js

```diff
@@ -895,6 +895,39 @@ function parseConfiguration(name, content, parsers) {
   };
 }
 
+/**
+ * Resolve referenced "extends" path in a configuration file
+ * using path.resolve() and require.resolve() as a fallback.
+ *
+ * @param {string} configFile Configuration file name.
+ * @param {string} referenceId Referenced path to resolve.
+ * @returns {string} resolved path to file
+ */
+function resolveConfigExtends(configFile, referenceId) {
+  const configFileDirname = path.dirname(configFile);
+  const resolvedExtendsFile = path.resolve(configFileDirname, referenceId);
+
+  try {
+    if (fs.statSync(resolvedExtendsFile).isFile()) {
```

Comment:
I shared some thoughts on `isFile()` in https://github.com/DavidAnson/markdownlint/pull/342#discussion_r507223896. At the moment, the only resulting error code is `ENOENT`, which can be emitted later from `readFile` / `readFileSync`. 

The current idea is that the resolve function never throws (as before). We could change this, but this would require more design work and could possibly make the change more breaking for the existing users / tools.

Reply:
Ok